Problem1.playground/Contents.swift

```diff
@@ -0,0 +1,130 @@
+/* Task 1.
+ Create a program to print the mean, median and mode of N numbers.
+ You should enter number N and then numbers, don't use standard functions, write your own.
+ */
+
+// Get a new array of fixed length
+func getArray(enterNumberN numberN: Int, ofNumbers numbers: [Int]) -> [Int] {
+    var arrayForPrint = [Int]()
+    arrayForPrint.reserveCapacity(numberN)
+
+    if numbers.isEmpty || numberN == 0 {
+        print("Empty array. Enter correct data!")
+    } else {
+        for (index, value) in numbers.enumerated() {
+            if index < numberN {
+                arrayForPrint.append(value)
+            }
+        }
+    }
+    
+    return arrayForPrint
+}
+
+func mean(ofNumbers numbers: [Int]) -> Double? {
+    var sum = 0
+    var mean: Double?
+    
+    if numbers.isEmpty {
+        mean = nil
+    } else {
+        for value in numbers {
+            sum += value
+        }
+        mean = Double(sum) / Double(numbers.count)
+    }
+    
+    return mean
+}
+
+func median(ofNumbers numbers: [Int]) -> Double? {
+    var median: Double?
+    
+    if numbers.isEmpty {
+        median = nil
+    } else {
+        let sortedArray = numbers.sorted { $0 < $1 }
+        let middleIndex = sortedArray.count / 2
+
+        if sortedArray.count.isMultiple(of: 2) {
+            median = Double(sortedArray[middleIndex] + sortedArray[middleIndex - 1]) / 2.0
+        } else {
+            median = Double(sortedArray[middleIndex])
+        }
+    }
+
+    return median
+}
+
+func mode(ofNumbers numbers: [Int]) -> [Double?] {
+    var mode = [Double?]()
+    
+    if numbers.isEmpty {
+        mode = [nil]
+    } else {
+        let sortedArray = numbers.sorted { $0 < $1 }
+        // Key-number: Value-count
+        var pairsOfNumbers = [Double: Double]()
```

Comment:
```suggestion
        var counters = [Double: Double]()
```

or 

```suggestion
        var modeCounters = [Double: Double]()
```

It will tell the purpose of variable and then you can remove the comment.

Reply:
Done.